services/portal/exportToWorkspace/exportToWorkspaceTasks.js

```diff
@@ -0,0 +1,113 @@
+  async mountLatestManifestInJupyterNotebook() {
+    const output = await within({ frame: exportToWorkspaceProps.workspaceIFrameClass }, () => {
+      I.waitForVisible(exportToWorkspaceProps.mainWorkspaceAppSelector, 600);
+      I.click(exportToWorkspaceProps.newDropDownButtonSelector);
+      I.click(exportToWorkspaceProps.newPython3ButtonSelector);
+      I.waitForVisible(exportToWorkspaceProps.pythonNotebookPanelSelector, 60); // wait for new notebook to show up
+      exportToWorkspaceProps.mountManifestCode.forEach((line) => {
+        I.pressKey(line);
+        I.pressKey('Enter');
+      });
+      I.wait(3); // wait for save button becomes clickable
+      I.click(exportToWorkspaceProps.runNotebookButtonSelector);
+      I.waitForVisible(exportToWorkspaceProps.pythonNotebookOutputAreaSelector, 60);
+      return I.grabTextFrom(exportToWorkspaceProps.pythonNotebookOutputAreaSelector);
+    });
+    return output;
+  },
+
+  /* Get the mounted manifest filename from python output */
+  async grabManifestFilename() {
+    const result = await I.grabTextFrom(exportToWorkspaceProps.exportToWorkspaceFooterClass);
+    const splittedResult = result.split('File Name: ');
+    if (splittedResult.length === 2) {
+      return splittedResult[1];
+    }
+    return '';
+  },
+
+  /* Step out from notebook and go back to workspace page */
+  async backToWorkspace() {
+    within({ frame: exportToWorkspaceProps.workspaceIFrameClass }, () => {
+      I.waitForVisible(exportToWorkspaceProps.saveNotebookButtonSelector, 10);
+      I.click(exportToWorkspaceProps.saveNotebookButtonSelector);
+      I.wait(3);
+      I.click(exportToWorkspaceProps.backToWorkspaceLinkSelector);
+    });
+  },
+
+  /* Remove all notebooks from workspace */
+  async deleteAllJupyterNotebooks() {
+    within({ frame: exportToWorkspaceProps.workspaceIFrameClass }, async () => {
+      I.waitForVisible(exportToWorkspaceProps.selectAllCheckboxSelector, 30);
+      I.click(exportToWorkspaceProps.selectAllCheckboxSelector);
+      I.waitForVisible(exportToWorkspaceProps.selectAllDropDownButtonSelector, 30);
+      I.click(exportToWorkspaceProps.selectAllDropDownButtonSelector);
+      I.waitForVisible(exportToWorkspaceProps.selectAllNotebookButtonSelector, 10);
+      I.click(exportToWorkspaceProps.selectAllNotebookButtonSelector);
+      I.waitForVisible(exportToWorkspaceProps.deleteButtonXPath, 10);
+      I.click(exportToWorkspaceProps.deleteButtonXPath);
+      I.wait(3); // wait for modal dialog to show
+      await this.checkAndAcceptJupyterDialogIfVisible();
+    });
+  },
+
+  /* Check if there is a modal dialog from Jupyther, if yes, accept it */
+  async checkAndAcceptJupyterDialogIfVisible() {
+    const ct = await I.grabNumberOfVisibleElements(exportToWorkspaceProps.modalDialogSelector);
+    if (ct === 1) {
+      const bt = await I.grabNumberOfVisibleElements(exportToWorkspaceProps.deleteConfirmButtonXPath);
+      if (bt === 1) {
```

Comment:
same here, what happens if not 1?

Reply:
same here, both `ct` and `bt` can only be 0 or 1. If `ct` is 0 then no dialog is showing up, hence I think it should do nothing.

Reply:
in which case would the dialog not be visible? i think maybe this should either be 2 separate functions, or this function should have an expected behavior

Reply:
In this case the dialog should always be visible. I was trying to make this method to be more generalize so if later we have a sequence that may not always have a visible dialog and we need to check before accepting we can use this as well.

Reply:
mm i see what you mean but in my opinion this function should just be `acceptJupyterDialog()` and you can do `I.see` and `I.click`, unless you have an idea of what use case would have a dialog that is sometimes visible, sometimes not

Reply:
agreed! I'll change this logic

Reply:
actually I'll remove this task and merge the `see` and `click` delete confirmation dialogic into `deleteAllJupyterNotebooks()`